Lib/fractions.py

```diff
@@ -244,7 +243,8 @@ def __new__(cls, numerator=0, denominator=None):
                 self._denominator = numerator.denominator
                 return self
 
-            elif isinstance(numerator, (float, Decimal)):
+            elif (isinstance(numerator, numbers.Number) and
```

Comment:
Performance nitpick. isinstance checks for abc classes are slow:
```
$ ./python -m timeit -r20 -s 'from numbers import Number as N; import numbers;from decimal import Decimal as D;a=1.1' 'isinstance(a, numbers.Number)'
500000 loops, best of 20: 831 nsec per loop
$ ./python -m timeit -r20 -s 'from numbers import Number as N; import numbers;from decimal import Decimal as D;a=1.1' 'isinstance(a, float)'
5000000 loops, best of 20: 75.3 nsec per loop
```

In main:
```
$ ./python -m timeit -r20 -s 'from fractions import Fraction as F;a=1.1' 'F(a)'
50000 loops, best of 20: 6.84 usec per loop
```
In this branch:
```
$ ./python -m timeit -r20 -s 'from fractions import Fraction as F;a=1.1' 'F(a)'
50000 loops, best of 20: 8.42 usec per loop
```

```suggestion
            elif (isinstance(numerator, (float, Decimal, numbers.Number)) and
```

Reply:
I would like to avoid importing `decimal`. This is a side benefit of this PR. But adding a fast path for float may be worthwhile.